reduce/src/cli.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+from pathlib import Path
+from .types import Command
+
+
+class Colors:
+    HEADER = "\033[95m"
+    OKBLUE = "\033[94m"
+    OKCYAN = "\033[96m"
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    ENDC = "\033[0m"
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
+    MAGENTA = "\u001b[35m"
+
+
+def error(msg: str) -> None:
+    print(f"{Colors.FAIL}[ERROR]{Colors.ENDC} {msg}")
+    exit(1)
+
+
+def warn(msg: str, ret_only: bool = False) -> str:
+    msg = f"{Colors.WARNING}[WARNING]{Colors.ENDC} {msg}"
+    if ret_only:
+        return msg
+    print(msg)
+    return msg
```

Comment:
```python
if not ret_only:
    print(msg)
return msg
```

?

Reply:
This is a gimmicky hack to get the formatting from the CLI api into a use-case where we don't want it printed (a.k.a: returned only). The only use case currently is for `input()` where I want to display a nicely-styled (via cli.py) but don't want to rewrite the code for styling. In that case, I just pass `ret_only=True` so that I can fetch the formatted snippet and display it in `input()` to the user.

Thoughts on this? Any alternatives?

Reply:
I was not thinking about the functionality of the function (I am sure it is fine), my comment was more around de-duplicating the `return msg`. Sorry, I should have made that a little clearer :)

Reply:
Ah, great style suggestion. Fixing now over at https://github.com/ClangBuiltLinux/reduce/pull/1 as [this](https://github.com/ClangBuiltLinux/misc-scripts/pull/4) PR will no longer receive code updates.